client/src/components/button.tsx

```diff
@@ -8,7 +8,20 @@ import rebuild from "../assets/rebuild.svg";
 import { StyleSheet, css } from "aphrodite";
 import { Link } from "react-router-dom";
 
-export const Button = ({
+interface OwnProps {
+  id?: string;
+  text: string | undefined;
+  type: string;
```

Comment:
так Typescript действительно будет помогать
```suggestion
  type: 'control' | 'action' | 'base';
```